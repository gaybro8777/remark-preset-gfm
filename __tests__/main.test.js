/**
 * @file main
 * @author imcuttle
 * @date 2018/4/4
 */

const remark = require('remark')
const html = require('remark-html')
const fs = require('fs')

const gfm = require('../')
const { fixture } = require('./helper')

function read(name) {
  return fs.readFileSync(fixture(name)).toString()
}

function getChildren(ast) {
  return ast && ast.children
}

describe('remarkPresetGfm', function() {
  it('should autolink-headings', function() {
    expect(
      remark()
        .use(gfm)
        .use(html)
        .use(require('../strip-position'))
        .processSync(read('autolink-headings.md'))
        .toString()
    ).toMatchSnapshot()
  })

  it('should gemoji-to-emoji', function() {
    expect(
      remark()
        .use(gfm)
        .use(require('../strip-position'))
        .processSync(read('gemoji-to-emoji.md'))
        .toString()
    ).toMatchSnapshot()
  })

  it('should frontmatter', function() {
    expect(
      remark()
        .use(gfm)
        .use(require('../strip-position'))
        .use(html)
        .processSync(read('frontmatter.md'))
        .toString()
    ).toMatchSnapshot()
  })

  it('should github-break', function() {
    expect(
      remark()
        .use(gfm)
        .use(require('../strip-position'))
        .use(html)
        .processSync(read('github-break.md'))
        .toString()
    ).toMatchInlineSnapshot(`
"<p>newline<br>
newline<br>
endline</p>
"
`)
  })

  it('should highlight.md', function() {
    expect(
      remark()
        .use(gfm)
        .use(require('../strip-position'))
        .use(html)
        .processSync(read('highlight.md'))
        .toString()
    ).toMatchInlineSnapshot(`
"<pre><code class=\\"hljs language-javascript\\"><span class=\\"hljs-built_in\\">console</span>.log(<span class=\\"hljs-number\\">123</span>)</code></pre>
<pre><code>function a() {}
</code></pre>
"
`)
  })
})
