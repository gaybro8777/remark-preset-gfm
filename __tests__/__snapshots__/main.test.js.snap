// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remarkPresetGfm should autolink-headings 1`] = `
"<h1 id=\\"lorem-ipsum-\\"><a href=\\"#lorem-ipsum-\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>Lorem ipsum 😪</h1>
<h2 id=\\"dolorsitamet\\"><a href=\\"#dolorsitamet\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>dolor—sit—amet</h2>
<h3 id=\\"consectetur--adipisicing\\"><a href=\\"#consectetur--adipisicing\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>consectetur &#x26; adipisicing</h3>
<h4 id=\\"elit\\"><a href=\\"#elit\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>elit</h4>
<h5 id=\\"elit-1\\"><a href=\\"#elit-1\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>elit</h5>
<h3 id=\\"你好-world\\"><a href=\\"#%E4%BD%A0%E5%A5%BD-world\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>你好 world</h3>
"
`;

exports[`remarkPresetGfm should frontmatter 1`] = `
"
"
`;

exports[`remarkPresetGfm should gemoji-to-emoji 1`] = `
"👍 💨
"
`;
